### ISSUE TITLE
[java] ATFD calculation problem

### ISSUE THREAD

**Issue by EmileSonneveld** [pmd/pmd#1910]
**Affects PMD Version:** 
6.16.0
and 53fbbb5a2d8514cacb551350ad9ce8e860c794c0

**Rule:**
[GodClass](https://pmd.github.io/latest/pmd_rules_java_design.html#godclass)

**Description:**
Some external properties are not calculated when code is structured in a certain way. 

Example in: https://github.com/gulrich/HotDraw/blob/master/JHotDraw%20%5BJava%5D/src/org/jhotdraw/application/DrawApplication.java#L941

**Code Sample demonstrating the issue:**

```java

// Counts as 2 external properties:
//  Possible God Class (WMC=130, ATFD=105, TCC=0.631%)
FileFilter tmp = openDialog.getFileFilter();
showStatus("Not a valid file format: " + tmp.getDescription());

// Counts no external properties:
// Possible God Class (WMC=130, ATFD=107, TCC=0.631%)
showStatus("Not a valid file format: " + openDialog.getFileFilter().getDescription());
```

For me, both code fragments should give the same ATFD

**Running PMD through:** *CLI *

**Comment by EmileSonneveld** [comment]
That indeed solved my issue, thank you! :)
